doc/userdoc/styleguide/styleguide.rst

```diff
@@ -0,0 +1,566 @@
+The NEST documentation style guide
+==================================
+
+.. contents::
+  :local:
+
+Why do we have a style guide?
+-----------------------------
+
+This style guide was created to provide a single reference point for content
+creation in NEST. This helps ensure the NEST user-level documentation remains
+clear and consistent. The style choices we make here are meant to follow the
+major trends in technical writing for software.
+
+General guidelines about writing for NEST
+-----------------------------------------
+
+Writing documentation: a few tips
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- Be concise - Use short words and sentences; avoid unnecessary modifiers
+
+- Be consistent - Follow this style guide and relevant templates
+
+- Be specific - Avoid vague language; clarity is key
+
+- Focus your message - Start with what's most important, and keep text relevant
+  within given section
+
+- Write for your reader - Try to understand what your reader needs to know;
+  include context to what you're saying
+
+- Provide examples - Show don't tell; if you can use code snippets, screenshots,
+  or figures to illustrate a point, do so!
+
+- Avoid terms like simply, just, or easy - your knowledge does not always equal
+  your readers.  Don't make your reader feel bad because they -simply- didn't
+  understand something.
+
+- Write positively -  Use positive language rather than negative language
+
+Remember that the primary purpose of the documentation is to provide
+information to NEST users who are looking for it.
+
+Voice
+~~~~~~
+
+Voice is *how* we talk to people through writing. The voice encompasses
+substance, style, and tone.
+
+Our voice is
+
+- empathetic - we write thinking of who we are writing for and what their needs
+  are
+- relaxed - we write in natural, conversational way rather than an authoritative
+  way
+
+Use 'you' to indicate the reader and 'we' to indicate yourself (the writer and
+possibly the NEST team).
+
+The language we use
+~~~~~~~~~~~~~~~~~~~
+
+We follow spelling and grammar rules of American English.
+
+The markup language we use is reStructuredText. This includes documentation
+blocks in files written in Python and cpp.
+
+We avoid slang or jargon that is unnecessary or could be confusing to readers.
+
+We do, however, use terminology appropriate to NEST and neuroscience when
+specific and exact terms are needed, which is often. But explanations or
+references should be provided for clarity in introductory texts.
+
+If you have further questions about style not addressed here, see <<STANDARDIZED STYLE GUIDE>>
```

Comment:
@heplesser  This is an external guide that we could use as a reference; basically the idea is that we only keep a limited set of guidelines or rules and use a comprehensive source, such as the [Microsoft style guide](https://docs.microsoft.com/en-us/style-guide/welcome/), as an authority on questions that arise. For example, if someone wants to know if we should put spaces around an em dash - rather than adding it to our guide, we refer person to reference. I explicitly mention the Microsoft Style Guide because it is and will very likely continue to be well maintained and is a style guide for the computer screen. So the MS guide _or something similar_ may be  a better choice than say, the Chicago Manual or APA style guides.

This is just a suggestion - I saw other projects do this, and thought it could be useful, but it's certainly not a necessity.